web/app/scripts/views/map/layers-controller.js

```diff
@@ -251,154 +262,207 @@
             */
         }
 
-        /**
-         * Adds the map layers. Removes them first if they already exist.
-         *
-         */
+
         ctl.setRecordLayers = function() {
             if (!ctl.map) {
                 $log.error('Map controller has no map! Cannot add layers.');
                 return;
             }
-            $q.all([TileUrlService.recTilesUrl(ctl.recordType),
-                TileUrlService.recUtfGridTilesUrl(ctl.recordType),
-                TileUrlService.recHeatmapUrl(ctl.recordType),
-                BlackspotSets.query({
-                    'effective_at': FilterState.getDateFilter().maxDate,
-                    'record_type': ctl.recordType
-                }).$promise.then(function(blackspotSet) {
-                    var set = blackspotSet[blackspotSet.length - 1];
-                    if (set) {
-                        return TileUrlService.blackspotsUrl(set.uuid);
-                    }
-                    return undefined;
-                })
-            ]).then(function(tileUrls) {
-                var baseRecordsUrl = tileUrls[0];
-                var baseUtfGridUrl = tileUrls[1];
-                var baseHeatmapUrl = tileUrls[2];
-                var blackspotsUrl = tileUrls[3];
-                var defaultLayerOptions = {
-                    attribution: 'PRS',
-                    detectRetina: true
-                };
 
-                // remove overlays if already added
-                if (ctl.overlays) {
-                    angular.forEach(ctl.overlays, function(overlay) {
-                        ctl.map.removeLayer(overlay);
+            if(ctl.layerSwitcher){
+                getUrls()
+                    .then(updateLayerGroups);
```

Comment:
Not sure there needs to be a linebreak here since there's only one `.then` call.